src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -86,23 +89,41 @@ func (r *StatefulSetResource) runUpdate(
 		return nil
 	}
 
-	if err = r.updateRestartingStatus(ctx, true); err != nil {
-		return fmt.Errorf("unable to turn on restarting status in cluster custom resource: %w", err)
+	if !r.getRestartingStatus() {
+		log.V(logger.DebugLevel).Info("restarting is required. Updating Cluster restarting status")
+		if err = r.updateRestartingStatus(ctx, true); err != nil {
+			return fmt.Errorf("unable to turn on restarting status in cluster custom resource: %w", err)
+		}
+		log.V(logger.DebugLevel).Info("updating restarting status on pods")
+		if err = r.MarkPodsForUpdate(ctx); err != nil {
+			return fmt.Errorf("unable to mark pods for update: %w", err)
+		}
 	}
 
+	log.V(logger.DebugLevel).Info("updating statefulset")
 	if err = r.updateStatefulSet(ctx, current, modified); err != nil {
 		return err
 	}
 
+	log.V(logger.DebugLevel).Info("checking if cluster is healthy")
 	if err = r.isClusterHealthy(ctx); err != nil {
 		return err
 	}
 
+	log.V(logger.DebugLevel).Info("performing rolling update")
 	if err = r.rollingUpdate(ctx, &modified.Spec.Template); err != nil {
 		return err
 	}
 
-	// Update is complete for all pods (and all are ready). Set restarting status to false.
+	// if !ALL_PODS_ARE_UPDATED() {
+	// 	return &RequeueAfterError{
+	// 		RequeueAfter: wait.Jitter(r.decommissionWaitInterval, decommissionWaitJitterFactor),
+	// 		Msg:          "waiting for all pods to be updated",
+	// 	}
+	// }
+
+	// If update is complete for all pods (and all are ready). Set restarting status to false.
+	log.V(logger.DebugLevel).Info("pod update complete, resetting restarting status on Cluster")
```

Comment:
```suggestion
	log.V(logger.DebugLevel).Info("pod update complete, set Cluster restarting status to false")
```

Reply:
reset is false. set is true. This is the definition.

Reply:
For me it's not easy to understand.